src/client/components/cube-view/cube-view.tsx

```diff
@@ -0,0 +1,295 @@
+'use strict';
+require('./cube-view.css');
+
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { List } from 'immutable';
+import { TimeRange } from 'plywood';
+import { DragManager } from '../../utils/drag-manager/drag-manager';
+import { Colors, Clicker, DataSource, Dimension, Essence, Filter, Stage, Manifest, Measure,
+  SplitCombine, Splits, TimePreset, VisStrategy, VisualizationProps} from '../../../common/models/index';
+// import { ... } from '../../config/constants';
+
+import { DimensionMeasurePanel } from '../dimension-measure-panel/dimension-measure-panel';
+import { FilterTile } from '../filter-tile/filter-tile';
+import { SplitTile } from '../split-tile/split-tile';
+import { VisSelector } from '../vis-selector/vis-selector';
+import { ManualFallback } from '../manual-fallback/manual-fallback';
+import { DropIndicator } from '../drop-indicator/drop-indicator';
+import { PinboardPanel } from '../pinboard-panel/pinboard-panel';
+
+import { visualizations } from '../../visualizations/index';
+
+export interface CubeViewProps extends React.Props<any> {
+  maxFilters?: number;
+  maxSplits?: number;
+  essence: Essence;
+}
+
+export interface CubeViewState {
+  essence?: Essence;
+  visualizationStage?: Stage;
+  menuStage?: Stage;
+  dragOver?: boolean;
+}
+
+export class CubeView extends React.Component< CubeViewProps, CubeViewState> {
```

Comment:
What is up with the space in `< CubeViewProps` was that added by the auto code formatter?